quantum/wpm.c

```diff
@@ -71,7 +71,7 @@ __attribute__((weak)) uint8_t wpm_regress_count(uint16_t keycode) {
 void update_wpm(uint16_t keycode) {
     if (wpm_keycode(keycode)) {
         if (wpm_timer > 0) {
-            uint16_t latest_wpm = 60000 / timer_elapsed(wpm_timer) / WPM_ESTIMATED_WORD_SIZE;
+            int16_t latest_wpm = 60000 / timer_elapsed(wpm_timer) / WPM_ESTIMATED_WORD_SIZE;
```

Comment:
With that `60000`, is `int16_t` large enough to handle the range of values given by the RHS? (also, what happens if `timer_elapsed(wpm_timer)` is 0?)

@firetech, mind taking a look at this, as you were the last person to touch this line, apparently fixing an *overflow* issue in #13128.

Reply:
`int16_t` might be a bit limited. If I press two keys at the same time on my keyboard, the RHS ends up around 12.000... However, I mainly set it to an `uint16_t` to be able to clamp such insane values down to 255 without relying (too much) on overflow, so `int16_t` should work, but might cause wrong calculations (i.e. somewhat random values due to overflow) in some cases.

That said, the if statement following this line should probably be altered to detect negative values? I don't entirely understand what this change solves, sounds like C magic that is a bit over my head.

Reply:
Max value should be 12k so that's fine in a signed int16 and none of the values in that calculation should be negative unless someone defines WPM_ESTIMATED_WORD_SIZE to be negative which would be strange.

Reply:
It solves the issue with the subtraction of 2 unsigned ints below it which is an unsigned result. When latest > current it causes unsigned underflow which makes the WPM count useless.